pkg/server/api/svid/v1/service_test.go

```diff
@@ -921,10 +926,206 @@ func TestServiceBatchNewX509SVID(t *testing.T) {
+
+	for _, tt := range []downstreamCaTest{
+		{
+			name:           "Malformed CSR",
+			rateLimiterErr: nil,
+			err:            "malformed CSR: asn1: structure error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.InvalidArgument,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "Rate Limiter Err",
+			rateLimiterErr: status.Error(codes.Internal, "rate limit error"),
+			err:            "rate limit error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "Unauthorized",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.NotFound,
+			fetcherErr:     "",
+			entry:          downstreamEntryUnauthorized,
+		},
+		{
+			name:           "Nonexistent entry",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.NotFound,
+			fetcherErr:     "",
+			entry:          &types.Entry{Id: "non-existent-entry"},
+		},
+		{
+			name:           "Entry fetch error",
+			rateLimiterErr: nil,
+			err:            "failed to fetch registration entries",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.Internal,
+			fetcherErr:     "Entry fetch error",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "FailDataStore",
+			rateLimiterErr: nil,
+			err:            "bundle not found",
+			failSigning:    false,
+			csrTemplate:    &x509.CertificateRequest{},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+			failDataStore:  true,
+		},
+		{
+			name:           "CertRequestWrongTd",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			failDataStore:  false,
+			csrTemplate: &x509.CertificateRequest{
+				URIs: []*url.URL{workloadID.URL()},
+			},
+			fetcherErr: "",
+			entry:      downstreamEntryWrongTd,
+			code:       codes.NotFound,
+		},
+		{
+			name:           "SuccessfulCARequest",
+			rateLimiterErr: nil,
+			err:            "",
+			failSigning:    false,
+			failDataStore:  false,
+			csrTemplate: &x509.CertificateRequest{
+				// Right now, the request will succeed even if URIs is empty. This is
+				// probably not the right behavior.
+				URIs: []*url.URL{workloadID.URL()},
+			},
+			fetcherErr: "",
+			entry:      downstreamEntry,
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+			test.ef.err = tt.fetcherErr
+			if !tt.failDataStore {
+				_, err := test.ds.AppendBundle(context.TODO(), &datastore.AppendBundleRequest{
```

Comment:
nit: `context.TODO()` is typically used when you expect to have a context threaded through at some point but don't have one available now. This isn't the case for tests though. `context.Background()` is more appropriate.